### ISSUE TITLE
Modernize Field interfaces

### ISSUE THREAD

**Issue by dcwhite** [SCIInstitute/SCIRun#1359]
- [ ] Use make_shared/make_unique instead of new
- [ ] Use `nullptr` instead of 0
- [ ] Add override keywords
- [ ] Fix remaining Resharper warnings

**Comment by dcwhite** [comment]
Needs a better field testing suite.

**Comment by dcwhite** [comment]
## Current state analysis (2026-06-12)

### `nullptr` instead of 0/NULL — ✅ DONE
Zero `NULL` or pointer-zero usages remain in the Field directory.

---

### `override` keywords — mostly done, ~61 remaining

| File | Missing overrides |
|------|----------------:|
| `VMesh.h` | many (base-class pure-virtual declarations — not candidates) |
| `VFData.h` | 18 |
| `ScanlineMesh.h`, `LatVolMesh.h`, `ImageMesh.h`, `CurveMesh.h` | 6 each |
| `TriSurfMesh.h`, `TetVolMesh.h`, `QuadSurfMesh.h`, `PrismVolMesh.h`, `PointCloudMesh.h`, `HexVolMesh.h` | 5 each |
| several others | 1–3 each |

866 `override` annotations already exist. The mesh-type headers are the remaining work: each has ~5 virtual methods in derived classes missing the keyword.

---

### `make_shared`/`make_unique` — three tiers of difficulty

Zero `make_shared`/`make_unique` calls exist anywhere in the Field directory today.

| Category | Count | Difficulty |
|----------|------:|-----------|
| `reset(new SearchGridT<...>)` — already in smart ptr, just needs `make_unique` | 17 | **Easy** |
| `mesh_handle_type(new mesh_type())` in GenericField ctor | 1 | **Easy** |
| `vfield_ = new VGenericField<...>` | 2 | Medium — check declared type of `vfield_` |
| `new Mutex` / `new std::map` lazy-init in `Mesh.cc` | 21 | Medium — repeated pattern, could use static locals or `call_once` |
| `new TypeDescription(...)` | 87 | **Hard** — static registry pattern; TypeDescription owns these via a static table, not RAII |
| `clone()` / `maker()` returning `Persistent*` | 25 | **Hard** — tied to serialization/Persistent factory framework; return type can't change without broader refactor |

---

### Resharper warnings
Still blocked on license (#1359 comment). Cannot assess without it.

---

### Suggested next steps
1. Mechanical `override` pass over the non-`VMesh.h` mesh headers (~50 sites, low risk)
2. Swap the 17 `reset(new SearchGridT<...>)` calls to `= std::make_unique<...>(...)`
3. Fix `GenericField` ctor and `vfield_` allocation (~3 sites)
4. Decide whether to tackle `TypeDescription`/`Persistent` new calls as a separate issue
